iree/turbine/kernel/wave/index_sequence_analysis.py

```diff
@@ -265,10 +265,48 @@ def has_gpr_offsets(node: fx.Node) -> bool:
             custom.graph.erase_node(custom.fx_node)
 
 
+def combine_derived_index(
+    src_index: Optional[dict[IndexSymbol, IndexSequence]],
+    dst_index: dict[IndexSymbol, IndexSequence],
+) -> dict[IndexSymbol, IndexSequence]:
+    if src_index is None:
+        return dst_index
+
+    new_index = copy(src_index)
+    for dim, new_idx in dst_index.items():
+        assert dim in src_index, f"Dim {dim} not in index {src_index}"
+        old_idx = src_index[dim]
+        if old_idx == new_idx:
+            continue
+
+        assert (
+            old_idx.start == 0 or old_idx.start == old_idx.start
+        ), f"Index conflict: {old_idx} and {new_idx}"
+        new_index[dim] = new_idx
+
+    return new_index
+
+
+def set_derived_index(trace):
+    sources = trace.walk(lambda node: isinstance(get_custom(node), (Read, Write)))
```

Comment:
This is fine, but just wondering, did you try adding this to propagate_index()? I think these can be combined together.